test/he/test_pmtud.c

```diff
@@ -0,0 +1,269 @@
+    TEST_ASSERT_EQUAL(1, call_counter);
+  }
+}
+
+void test_he_internal_pmtud_handle_probe_ack_from_searching_big_step(void) {
+  test_handle_probe_ack_from_searching(1212, true);
+}
+
+void test_he_internal_pmtud_handle_probe_ack_from_searching_small_step(void) {
+  test_handle_probe_ack_from_searching(1212, false);
+}
+
+void test_he_internal_pmtud_handle_probe_ack_from_searching_to_search_complete(void) {
+  test_handle_probe_ack_from_searching(MAX_PLPMTU, true);
+}
+
+void test_he_internal_pmtud_handle_probe_timeout_try_again(void) {
+  conn.state = HE_STATE_ONLINE;
+  conn.pmtud_state = HE_PMTUD_STATE_SEARCHING;
+  conn.pmtud_probe_count = 1;
+  conn.pmtud_time_cb = pmtud_time_cb;
+  conn.ping_next_id = 43;
+  conn.pmtud_probing_size = 1212;
+
+  // It should send probe again when probe count hasn't reached MAX_PROBES
+  EXPECT_HE_INTERNAL_SEND_MESSAGE();
+
+  TEST_ASSERT_EQUAL(HE_SUCCESS, he_internal_pmtud_handle_probe_timeout(&conn));
+
+  TEST_ASSERT_EQUAL(2, conn.pmtud_probe_count);
+  TEST_ASSERT_EQUAL(43, conn.pmtud_probe_pending_id);
+  TEST_ASSERT_EQUAL(44, conn.ping_next_id);
+}
+
+void test_he_internal_pmtud_handle_probe_timeout_confirm_based_failed(void) {
+  conn.state = HE_STATE_ONLINE;
+  conn.pmtud_state = HE_PMTUD_STATE_BASE;
+  conn.pmtud_probe_count = 3;
+
+  // Probe count reached MAX_PROBES,
+  // it should call he_internal_pmtud_confirm_base_failed if current state is Base.
+  TEST_ASSERT_EQUAL(HE_SUCCESS, he_internal_pmtud_handle_probe_timeout(&conn));
+
+  // The new state should be Error
+  TEST_ASSERT_EQUAL(HE_PMTUD_STATE_ERROR, conn.pmtud_state);
+
+  // The probe count should be reset to 0
+  TEST_ASSERT_EQUAL(0, conn.pmtud_probe_count);
+}
+
+void test_he_internal_pmtud_handle_probe_timeout_search_completed(void) {
+  conn.state = HE_STATE_ONLINE;
+  conn.pmtud_state = HE_PMTUD_STATE_SEARCHING;
+  conn.pmtud_probe_count = 3;
+  conn.pmtud_probing_size = 1212;
+  conn.pmtud_is_using_big_step = false;
+
+  // Probe count reached MAX_PROBES,
+  // it should call he_internal_pmtud_confirm_base_failed if current state is Base.
+  TEST_ASSERT_EQUAL(HE_SUCCESS, he_internal_pmtud_handle_probe_timeout(&conn));
+
+  // The new state should be Search Complete
+  TEST_ASSERT_EQUAL(HE_PMTUD_STATE_SEARCH_COMPLETE, conn.pmtud_state);
+
+  // The effective mtu should be set
+  TEST_ASSERT_EQUAL(1212 - PMTUD_PROBE_SMALL_STEP, conn.effective_pmtu);
+
+  // The probe count should be reset to 0
+  TEST_ASSERT_EQUAL(0, conn.pmtud_probe_count);
+}
+
+void test_he_internal_pmtud_handle_probe_timeout_blackhole_detected(void) {
+  conn.state = HE_STATE_ONLINE;
+  conn.pmtud_state = HE_PMTUD_STATE_SEARCH_COMPLETE;
+  conn.pmtud_probe_count = 3;
+  conn.pmtud_probing_size = 1212;
+  conn.pmtud_is_using_big_step = false;
+  conn.pmtud_time_cb = pmtud_time_cb;
+
+  EXPECT_HE_INTERNAL_SEND_MESSAGE();
+
+  // Probe count reached MAX_PROBES,
+  // it should call he_internal_pmtud_confirm_base_failed if current state is Base.
```

Comment:
I don't think `// it should call he_internal_pmtud_confirm_base_failed if current state is Base.` relevant for this test case?